extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/ClassNames.java

```diff
@@ -465,4 +465,22 @@ private static DotName createConstant(String fqcn) {
             "io.quarkus.hibernate.orm.deployment.HibernateUserTypeProcessor");
 
     public static final DotName GRAAL_VM_FEATURES = createConstant("io.quarkus.hibernate.orm.deployment.GraalVMFeatures");
+
+    public static final List<DotName> SERVICE_PROVIDERS = List.of(
+            // Accessed in org.hibernate.query.sqm.internal.SqmCriteriaNodeBuilder.<init>
+            createConstant("org.hibernate.query.criteria.spi.CriteriaBuilderExtension"),
+            // Accessed in io.quarkus.hibernate.orm.runtime.customized.QuarkusIntegratorServiceImpl.<init>
+            createConstant("org.hibernate.integrator.spi.Integrator"),
```

Comment:
`QuarkusIntegratorServiceImpl` should only be used during static init, as far as I can see:

https://github.com/quarkusio/quarkus/blob/0a1776e7f6a3a18576faa55e832bde27926a6c1e/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/RecordableBootstrapFactory.java#L40-L48

https://github.com/quarkusio/quarkus/blob/b4b93ada873b74250112ec0f70362b17af59c3df/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java#L132

So I wonder why it needs to be registered... ?

Reply:
You are right, it's not needed.